students/rhamersky/Lesson_5/mailroom_part3.py

```diff
@@ -0,0 +1,326 @@
+#      Update variables to snake_case. Removed unnecessary 'continues' and 'returns'.
+#      Changed 'bool' variable to 'bool_flag'. Updated misleading function names.
+#      Reduced the amount of unnecessary print() statements.
+
+# -----Data Section-----
+import os
+str_name = None     # --> User's name
+int_option = int()  # --> User's option.
+tup_donor_names = (["William Gates, III",1,10000.0],["Mark Zuckerberg",1,20000.0],
+                   ["Jeff Bezos",1,1200.0],["Paul Allen",1,5000.0],
+                   ["Mike Scott",1,25.0])  # --> Tuple that stores all donor's information
+str_salutation = ""   # --> Email salutation
+str_body = ""         # --> Email body
+str_valediction = ""  # --> Email valediction
+
+
+# -----Process Section-----
+#  Methods defined in the menu class are related to general menu checks or options based on user inputs.
+
+class menu(object):
+    @staticmethod
+    def menu_options():
+        '''
+        Try and except block used on User's option choice to check for integers.
+        :return: Returns int data (1-4) which is used to determine what menu option will be executed.
+        '''
+        while True:
+            try:
+                int_choice = int(input(str_name + ", please select an option above. "))
+            except ValueError:
+                print(str_name + ", not a valid option. \n")
+            # Verify that the user selected an option 1 through option 4.
+            else:
+                if 1 <= int_choice <= 4:
+                    break
+                else:
+                    print("Integer selected is not a valid option. "
+                          "Please, enter an integer and corresponds to the options given. \n")
+        return int_choice
+############## END OF MENU CLASS ######################
+
+
+# Methods in the sub_menu class will execute code based off of the user's sub-menu choice.
+
+
+class sub_menu(object):
+    @staticmethod
+    def send_thankyou():
+        '''
+        This method will create a list of donors, allow new donors to be added, store donations, and print a thank you
+        statement to send to the donor by mail.
+        :return:
+        '''
+        bool_flag = True  # --> Used as a flag.
+        donor_name = input("Please, enter full name of donor. ").title()
+
+        # Creates a donor list.
+        while donor_name == "List":
+            # --> Verifies that there is at least a donor is stored as a list row in the table tuple.
+            if len(tup_donor_names) > 0:
+                donor_list_creation()  # --> Creates a list of donors
+                bool_flag = False  # --> Is used as a flag on the while loop below
+            else:
+                print("\n"+"There are no donors in list.")  # --> Let's the user know that a donor list does not exist.
+                bool_flag = False  # --> Is used as a flag on the while loop below.
+            break
```

Comment:
why is this a while loop if breaking out of it after single run?

Reply:
Oh it should be an if statement. When I originally was writing this code I wanted the program to give the user another shot at putting in a user prior to calling for a list that was empty. As the program was originally designed for an empty donors list and the user would put in the information. Through revisions I made the decision to kick the user back to the main menu if the donor list is empty and the user tries to call for a list.